docs/docs/en/guide/task/flink.md

```diff
@@ -37,7 +37,7 @@ Flink task type, used to execute Flink programs. For Flink nodes:
 | Parallelism             | Used to set the degree of parallelism for executing Flink tasks.                                                                                                                                                                                                        |
 | Yarn queue              | Used to set the yarn queue, use `default` queue by default.                                                                                                                                                                                                             |
 | Main program parameters | Set the input parameters for the Flink program and support the substitution of custom parameter variables.                                                                                                                                                              |
-| Optional parameters     | Set the flink command options, such as `-D`, `-C`, `-yt`.                                                                                                                                                                                                               |
+| Optional parameters     | Set the flink command options, such as `-D`, `-C`, `-yt`, and support the substitution of custom parameter variables, such as `-Dyarn.application.name=${job_name}` custom parameter job _name will be replaced.                                                        |
```

Comment:
Is this custom parameter `job_name` setting by users with hard code? If so, this is meaningless. We should add separate field options for users to use instead of using custom parameters, which are suitable for parameter passing and global parameters.

Reply:
Not hard-coded, user custom parameter pass and replace, like main program parameters:
![image](https://github.com/user-attachments/assets/794fafe4-79fd-4b6b-96f8-209a6bb16eaa)

Reply:
As I said that Custom parameters are suitable for parameter passing between tasks in a workflow and global parameters. `yarn.application.name` is a parameter of task level with different functions. We should not add it here.

Reply:
I understand boundary with custom parameters, option parameter replacement is also specific to Flink Task.

Reply:
> option parameter replacement is also specific to Flink Task.

I don't think we should confuse usage of `custom parameters` with `task parameters`. Sorry, I'm -1 on this.

Reply:
> > option parameter replacement is also specific to Flink Task.
> 
> I don't think we should confuse usage of `custom parameters` with `task parameters`. Sorry, I'm -1 on this.

However, Flink task main parameters now allows custom parameters substitution.

Reply:
> Understand your idea, but i have a question: custom parameters can't be used as task parameters? Now flink task main parameters can be replaced with custom parameters.

I didn't say that `custom parameters can't be used as task parameters`. What I mean is `job_name` should be added into task parameters instead of custom parameters.

After take a look at `dev` branch, I found that the `job_name` already exists in task params right now with `org.apache.dolphinscheduler.plugin.task.flink.FlinkParameters#appName`. But it is only used in flink sql. We can just simpliy add it to the flink jar mode. And this issue can be easily solved.